fuzzer/src/main.rs

```diff
@@ -0,0 +1,223 @@
+        //*    Create state with previous data
+        //* ---------------------------------------
+
+        let mut state = CachedState::new(state_reader, Some(contract_class_cache));
+
+        //* ------------------------------------
+        //*    Create execution entry point
+        //* ------------------------------------
+
+        let calldata = [].to_vec();
+        let caller_address = Address(0000.into());
+        let entry_point_type = EntryPointType::External;
+
+        let exec_entry_point = ExecutionEntryPoint::new(
+            address.clone(),
+            calldata.clone(),
+            storage_entrypoint_selector.clone(),
+            caller_address,
+            entry_point_type,
+            Some(CallType::Delegate),
+            Some(class_hash),
+        );
+
+        //* --------------------
+        //*   Execute contract
+        //* ---------------------
+        let general_config = StarknetGeneralConfig::default();
+        let tx_execution_context = TransactionExecutionContext::new(
+            Address(0.into()),
+            Felt::zero(),
+            Vec::new(),
+            0,
+            10.into(),
+            general_config.invoke_tx_max_n_steps(),
+            TRANSACTION_VERSION,
+        );
+        let mut resources_manager = ExecutionResourcesManager::default();
+
+        let expected_key = calculate_sn_keccak("_counter".as_bytes());
+
+        let mut expected_accessed_storage_keys = HashSet::new();
+        expected_accessed_storage_keys.insert(expected_key);
+
+        let expected_call_info = CallInfo {
+            caller_address: Address(0.into()),
+            call_type: Some(CallType::Delegate),
+            contract_address: Address(1111.into()),
+            entry_point_selector: Some(storage_entrypoint_selector),
+            entry_point_type: Some(EntryPointType::External),
+            calldata,
+            retdata: [Felt::from_bytes_be(data_to_ascii(data).as_bytes())].to_vec(),
+            execution_resources: ExecutionResources::default(),
+            class_hash: Some(class_hash),
+            storage_read_values: vec![Felt::from_bytes_be(data_to_ascii(data).as_bytes())],
+            accessed_storage_keys: expected_accessed_storage_keys,
+            ..Default::default()
+        };
+        
+        assert_eq!(
+            exec_entry_point
+                .execute(
+                    &mut state,
+                    &general_config,
+                    &mut resources_manager,
+                    &tx_execution_context
+                )
+                .unwrap(),
+            expected_call_info
+        );
+        
+        assert!(!state.cache().storage_writes().is_empty());
+        assert_eq!(
+            state
+                .cache()
+                .storage_writes()
+                .get(&(address, expected_key))
+                .cloned(),
+            Some(Felt::from_bytes_be(data_to_ascii(data).as_bytes()))
+        );
```

Comment:
I think this should be intended an extra level.

Reply:
done!